whisper-finetune/inference_endpoint/score.py

```diff
@@ -0,0 +1,60 @@
+import numpy as np
+from onnxruntime import InferenceSession
+import os
+from transformers import WhisperProcessor
+import json 
+
+# Documentation: https://learn.microsoft.com/en-us/azure/machine-learning/how-to-deploy-online-endpoints
+# Troubleshooting: https://learn.microsoft.com/en-us/azure/machine-learning/how-to-troubleshoot-online-endpoints
+  
+# The init() method is called once, when the web service starts up.
+def init():  
+    global sess
+    global processor
+
+    processor = WhisperProcessor.from_pretrained("openai/whisper-small", language="Hindi", task="transcribe")
+
+    # The AZUREML_MODEL_DIR environment variable indicates  
+    # a directory containing the model file you registered.  
+    model_filename = "whisper-small/openai/whisper-small_beamsearch.onnx" 
+    model_path = os.path.join(os.environ['AZUREML_MODEL_DIR'], model_filename)  
+
+    sess = InferenceSession(model_path, providers=["CUDAExecutionProvider", "CPUExecutionProvider"])
+  
+  
+# The run() method is called each time a request is made to the scoring API.  
+def run(data):
+    json_data = json.loads(data)
+    audio = np.array(json_data["audio"])
+    
+    N_FRAMES = 3000
+    HOP_LENGTH = 160
+    SAMPLE_RATE = 16000
+    N_MELS = 80    
+    min_length = 0
+    max_length = 20
+    repetition_penalty = 1.0
+    beam_size = 1
+    NUM_RETURN_SEQUENCES = 1
+    input_shape = [1, N_MELS, N_FRAMES]
```

Comment:
May we have consistent naming of the variables?

Reply:
Done